docs/site/Inside-Loopback-Application.md

```diff
@@ -6,81 +6,146 @@ sidebar: lb4_sidebar
+
+## Exposing the API endpoints
+
+We can now start creating endpoints using the api specs in the above section. Developers typically create server-side "handlers" for each route. The handlers will have to accept incoming request, parse and validate.
+
+> LoopBack develepers can configure simple javascript methods as remote endpoints. Out-of-the-box parsing and validation is done using the provided specification.
+
+Remote methods are defined in [controller classes](https://loopback.io/doc/en/lb4/Controllers.html#routing-to-controllers) and configured as API endpoints with api `decorators`.
+
+> LoopBack has inbuilt decorators to indicate api specification and the expect arguments for the remote method.
+
+For Example,
+- `login()` method in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) class is defined as `/users/login` API endpoint.
+-  argument of method `findById` in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) is decorated with `@param.path.string('userId')` which means that the userId parameter in the URL path is passed into the method at runtime.
+
+## API Security
+
+> Security implementations in Loopback can be created as separate [Authentication components](https://loopback.io/doc/en/lb4/Loopback-component-authentication.html) and the `authenticate` decorator can be used to define the authentication strategy of a particular endpoint.
+
+For Example,
+- A [JWT Authentication Strategy](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/authentication-strategies/jwt-strategy.ts) is implemented with a name property `jwt`.
+- `printCurrentUser` remote method in [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) is decorated with `authenticate('jwt')` to indicate the API endpoint is authenticated with json web tokens.
+
+This helps in separation of security aspects of API endpoints from business logic and easy understanding of security specifications.
+
+## Business logic
+
+API implementation logic is separated between various layers in Loopback:
+
+- [Controllers](https://loopback.io/doc/en/lb4/Controllers.html) are used for orchestrating calls to `Service` and `Repository` classes.
 
-## Open API specification 
+- [Services](https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html) act as the API logical layer. They can connect with `repositories` and `datasources` to fetch data. Services interchangeably, are also used to invoke [external apis and services](https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html). This comes handy in a microservices environment.
```

Comment:
Just an FYI, the UserService and TokenService in loopback shopping cart app were named `services` but were not defined in the manner mentioned in https://loopback.io/doc/en/lb4/Calling-other-APIs-and-web-services.html (which shows how SOAP services are defined). Not sure if we need to point that out....